### ISSUE TITLE
Building a list of complex records with a form

### ISSUE THREAD

**Issue by allancaffee** [piceaTech/ember-rapid-forms#128]
First of all forgive me if this isn't the appropriate venue for asking questions. I'm currently trying to build up a form that would allow users to input structured data. I'm trying to build up a list of nested dictionaries like:

```javascript
{
  recipe: {
    // ...
    ingredients: [{
      amount: 2,
      unit: 'EACH',
      name: 'Potatoe'
    }, {
      amount: 1,
      unit: 'CUP',
      name: 'Milk'
    }, {
      amount: 1,
      unit: 'EACH',
      name: 'Egg'
    }
  },
  // ...
}
```

and I'm having a little trouble figuring out how to build up a list of complex objects with a web form. I was hoping for something similar to the `FieldList` functionality from WTForms (https://wtforms.readthedocs.io/en/latest/fields.html#wtforms.fields.FieldList) which basically lets users click a little plus sign to keep adding rows to the list.

I suppose worse case scenario I could put a free form text box in and parse it a line at a time, but I'd love to be able to make `unit` into a dropdown and have type-ahead completion for the `name` field to make things a little easier on mobile users.

Is this a pattern that you've seen before and does it make sense to implement something like this as part of rapid forms? Any pointers would be appreciated.

**Comment by spruce** [comment]
That is a nice feature. But sadly ember-rapid-forms doesn't support anything like type-aheads. (Nobody till now had the usecase) I also think this goes beyond what we want to support. I think the best way possible to go forward ist to build such a component yourself. In my opinion you don't even need ember-rapid-forms for that.

(Offtopic) Out of curiousity: How did you find this project?

**Comment by allancaffee** [comment]
You're the second hit on Ember Observer for forms https://emberobserver.com/?query=forms. And the first one I started really playing around with. Mostly because it seemed (and was) super simple to wire together a basic form and get it posting to my backend, and it seems like its under active development so I figured I would give it a shot.

I'll do a little more digging around in the docs for other existing component add-ons and if I can't find anything I'll try my hand at rolling my own. Thanks for the speedy response!
